sycl/doc/extensions/proposed/sycl_ext_intel_cache_controls.asciidoc

```diff
@@ -0,0 +1,201 @@
+or may be in a preliminary state.  The specification itself may also change in
+incompatible ways before it is finalized.  *Shipping software products should
+not rely on APIs defined in this specification.*
+
+
+== Backend support status
+
+This extension is not implemented.
+
+== Overview
+
+This extension introduces additional compile-time properties for
+the proposed `sycl::ext::oneapi::experimental::annotated_ptr` class to specify
+cache control information.
+
+The cache controls are a strong request that memory accesses through the
+pointer should use instructions with the specified cache controls.
+However, the implementation may choose a different cache control or none
+if the requested one is unsupported or for any other reason.
+
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_CACHE_CONTROLS` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Properties
+
+Below is a list of new compile-time constant properties supported with
+`annotated_ptr`.
+
+```c++
+namespace sycl::ext::intel::experimental {
+
+enum cache_control_types_read {
+    Cached, Uncached, Streaming, InvalidateAfterRead, ConstCached};
+enum cache_control_types_write {
+    Uncached, Streaming, WriteThrough, WriteBack};
+
+struct cache_control_read {
+  template<enum cache_control_types_read, int L>
+  using value_t = property_value<cache_control_read, enum control, std::integral_constant<int, L>>;
+};
+struct cache_control_write {
+  template<enum cache_control_types_write, int L>
+  using value_t = property_value<cache_control_write, enum control, std::integral_constant<int, L>>;
+};
+
+template<cache_control_types_read C, int L>
+inline constexpr cache_control_read::value_t<C, K> read_control;
+template<cache_control_types_write C, int L>
+inline constexpr cache_control_write::value_t<C, K> write_control;
+
+template<>
+struct is_property<cache_control_read> : std::true_type {};
+template<>
+struct is_property<cache_control_write> : std::true_type {};
+
+template<typename T, typename PropertyListT>
+struct is_property_of<
+  cache_control_read, annotated_ptr<T, PropertyListT>> : std::true_type {};
+template<typename T, typename PropertyListT>
+struct is_property_of<
+  cache_control_write, annotated_ptr<T, PropertyListT>> : std::true_type {};
+
+template<typename T, typename PropertyListT>
+} // namespace sycl::ext::intel::experimental
+```
```

Comment:
There should be a description of the cache level and a description about limitations.  I'd also like to document the behavior-modifying properties separately from the others.  After the synopsis, we could have paragraphs like this:

> Each of these properties takes a cache level parameter indicating which level of the cache hierarchy is affected.  Cache level 0 indicates the cache closest to the processing unit, cache level 1 indicates the next furthest cache level, etc.  It is legal to specify a cache level that does not exist on the target device, but the property will be ignored in this case.
>
> Note that a property specifies the cache behavior only for the indicated cache level.  In order to specify the behavior for multiple cache levels, multiple properties should be specified.
>
> It is legal to specify several different cache control properties in the same `annotated_ptr`.  However, all instances of `cache_control_read` must have different cache levels and all instances of `cache_control_write` must have difference cache levels.
>
> The cache control properties are divided into two categories: those that are hints and those that are assertions by the application.
>
> ==== Cache control hints
>
> These properties are hints requesting specific cache behavior when loading or storing to memory through the `annotated_ptr`.  These properties can affect the performance of device code, but they do not change the semantics.
>
> [Table describing all properties except `InvalidateAfterRead` and `ConstCached`]
>
> ==== Assertions by the application
>
> These properties are assertions by the application, promising that the application accesses memory in a certain way.  Care must be taken when using these properties because they can lead to undefined behavior if they are misused.
>
> [Table describing `InvalidateAfterRead` and `ConstCached`]

Reply:
Good stuff. Added this text and broke up table.